src/app/public/modules/modal/modal.component.ts

```diff
@@ -97,6 +115,9 @@ export class SkyModalComponent implements AfterViewInit {
     return this.config.helpKey;
   }
 
+  @Output()
+  public beforeClose = new EventEmitter<SkyModalBeforeCloseHandler>();
```

Comment:
It might be better to handle the `beforeClose` hook from the returned modal instance, so that the consumer can handle both the `beforeClose` and `closed` hooks in the same place. What were the challenges of implementing the following pattern?

```
const instance = this.modal.open(
  MyModalComponent
);

instance.beforeClosed.subscribe((handler) => {
  handler.triggerClose();
});

instance.closed.subscribe(...);
```

Reply:
I don't know of any immediately. Just hadn't considered it. I agree though it would make more sense to put it where the closed event is available

Reply:
Moved to the instance. I also added an option there so that instance.close calls can still bypass it if they want

Reply:
@Blackbaud-SteveBrush ^